src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt

```diff
@@ -249,5 +249,25 @@ class PetDaoTest @Autowired constructor(private val sql: DSLContext) {
 
     }
 
+    @Test
+    fun `count by type`() {
+        petDao.createPet(PetNoId("Jerry", "Dog", companyId, Date.valueOf(LocalDate.now()), null))
```

Comment:
You are creating same data over and over
you can add the PetNoId in the test class out of a specific test and reuse it any test you want